src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1477,7 +1693,9 @@ Middleware allows you to intercept requests and responses and inject behaviors d
 
 ### `onRequest()`
 
-A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It accepts two optional arguments: [context](#contextlocals) and [next()](#next). `onRequest()` must return a `Response`: either directly, or by calling `next()`.
+**Type:** `(context: APIContext, next: MiddlewareNext) => Promise<Response> | Response | Promise<void> | void`
+
+A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It accepts two optional arguments: [context](#context) and [next()](#next). `onRequest()` must return a `Response`: either directly, or by calling `next()`.
```

Comment:
Small nit regarding the initial wording here:

```suggestion
A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It receives two arguments: [context](#context) and [next()](#next). `onRequest()` must return a `Response`: either directly, or by calling `next()`.
```

Astro always passes the arguments to `onRequest` so they are not “optional arguments”. Whether a user uses them or not is up to them, but they will always be present if they do need them.